### PR TITLE
impl: add remaining universe_domain service demos

### PR DESCRIPTION
**Pull request by scotthart**
This adds the remaining service demos specified for the universe_domain feature. Additional auth related demos are next.

Demos now are built with cmake and execute as part of the quickstart-production builds.

part of the work for #13495

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13496)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -231,14 +231,19 @@ if (BUILD_TESTING)
...
-# the flag.
+# The examples and demos use exception handling to simplify the code. Therefore
+# they cannot be compiled when exceptions are disabled, and applications cannot
+# force the flag.
 cmake_dependent_option(
     GOOGLE_CLOUD_CPP_ENABLE_EXAMPLES "Compile the google-cloud-cpp examples."
     ON "GOOGLE_CLOUD_CPP_ENABLE_CXX_EXCEPTIONS;BUILD_TESTING" OFF)
 mark_as_advanced(GOOGLE_CLOUD_CPP_ENABLE_EXAMPLES)
 
+cmake_dependent_option(
```
Comment (dbolduc) [inline CMakeLists.txt:242]: I wouldn't make this a public option. I don't think demos are any different than examples. It feels like more of a way to hack the CI scripts.

**Review by dbolduc — CHANGES_REQUESTED** [review]

File: CMakeLists.txt [hunk 2]
```diff
@@ -231,7 +231,7 @@ if (BUILD_TESTING)
     include(FindGMockWithTargets)
 endif ()
 
-# The examples use exception handling to simplify the code. Therefore they
+# The examples  use exception handling to simplify the code. Therefore they
```
Comment (dbolduc) [inline CMakeLists.txt:234]: nit: seems like we introduced an extra space

File: google/cloud/universe_domain/CMakeLists.txt [hunk 3]
```diff
@@ -16,18 +16,72 @@
...
-            cmake -P "${PROJECT_SOURCE_DIR}/cmake/quickstart-runner.cmake"
-            $<TARGET_FILE:universe_domain_kms_demo> GOOGLE_CLOUD_PROJECT
-            GOOGLE_CLOUD_CPP_TEST_REGION)
-    set_tests_properties(universe_domain_kms_demo
-                         PROPERTIES LABELS "integration-test;quickstart")
+if (GOOGLE_CLOUD_CPP_ENABLE_EXAMPLES AND GOOGLE_CLOUD_CPP_ENABLE_CXX_EXCEPTIONS)
+    set(universe_domain_demos bigquery compute kms pubsub storage)
+    foreach (library ${universe_domain_demos})
+        if (${library} IN_LIST GOOGLE_CLOUD_CPP_ENABLE)
+
```
Comment (dbolduc) [inline google/cloud/universe_domain/CMakeLists.txt:23]: nit: remove blank line

**Review by dbolduc — APPROVED** [review]

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13496)* status: 0 of 9 files reviewed, 6 unresolved discussions (waiting on @dbolduc)

___
*[`google/cloud/universe_domain/CMakeLists.txt` line 23 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13496#gh-1465449891:-Nox1q2fB3y93Do1rFPb:ba3dfob) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/e0803f9818a8e9266e5c213ffb61f2461c9ef306/google/cloud/universe_domain/CMakeLists.txt#L23)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

nit: remove blank line
</blockquote></details>

Removed
___
*[`CMakeLists.txt` line 234 at r2](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13496#gh-1465449291:-Nox1rU58aoFOwiDJoRt:ba3dfob) ([raw file](https://github.com/googleapis/google-cloud-cpp/blob/e0803f9818a8e9266e5c213ffb61f2461c9ef306/CMakeLists.txt#L234)):*
<details><summary><i>Previously, dbolduc (Darren Bolduc) wrote…</i></summary><blockquote>

nit: seems like we introduced an extra space
</blockquote></details>

Removed

**Pull request merged**